extra_data/read_machinery.py

```diff
@@ -23,6 +23,7 @@
 DETECTOR_SOURCE_RE = re.compile(r'(.+\/(?:DET|CORR))\/(\d+)CH')
 
 DATA_ROOT_DIR = os.environ.get('EXTRA_DATA_DATA_ROOT', '/gpfs/exfel/exp')
+SW_ROOT_DIR = os.environ.get('EXTRA_DATA_SW_ROOT', '/gpfs/exfel/sw')
```

Comment:
I would recommend making this a function that looks up the environment variable, that way it's easier to test with `monkeypatch.setenv()` compared to patching every module where the variable is imported.